core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -1092,24 +1111,10 @@ oppia.factory('explorationStatesService', [
               };
             }
           ]
-        }).result.then(function(deleteStateName) {
-          _states.deleteState(deleteStateName);
-
-          changeListService.deleteState(deleteStateName);
-
-          if (editorContextService.getActiveStateName() === deleteStateName) {
-            editorContextService.setActiveStateName(
-              explorationInitStateNameService.savedMemento);
-          }
-
-          $location.path('/gui/' + editorContextService.getActiveStateName());
-          $rootScope.$broadcast('refreshGraph');
-          // This ensures that if the deletion changes rules in the current
-          // state, they get updated in the view.
-          $rootScope.$broadcast('refreshStateEditor');
-          // This state name is removed from gadget's visibilty settings.
-          explorationGadgetsService.handleStateDeletion(deleteStateName);
-        });
+        }).result.then(confirmDelete);
+      },
+      confirmDelete: function(deleteStateName) {
+        return confirmDelete(deleteStateName);
```

Comment:
Can we call the "private" function something different in order to avoid confusion? Maybe something like reallyConfirmDelete (a bit corny, but it works).